web/html/javascript/ajax.js

```diff
@@ -0,0 +1,20 @@
+var onAjaxRequestError = function (res, status, error) {
+  console.log("Error processing ajax request...")
+  console.log(res.responseText, status, error)
+}
+
+var ajax_post = function(url, data, onSuccess, onError, contentType = false) {
```

Comment:
If I understand correctly, you created this file since you can't import the network layer in regular Javascript files, right? The best option would of course be to have those files become part of the Typescript codebase instead, the second best option would be to instead make the network layer available globally and annotate why it's global.

The type annotations for making things global is `declare global { ... }`, you can find an example in `web/html/src/core/debug/index.ts` for example.  

Withe either one of those options we avoid having two slightly different network layer implementations.